packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts

```diff
@@ -0,0 +1,352 @@
+              ScheduleExpression: {
+                Ref: 'CloudWatchRule',
+              },
+              State: 'ENABLED',
+              Targets: [
+                {
+                  Arn: { 'Fn::GetAtt': ['LambdaFunction', 'Arn'] },
+                  Id: {
+                    Ref: 'LambdaFunction',
+                  },
+                },
+              ],
+            },
+          };
+          // append permissions to invoke lambda via cloiudwatch to CFN file
+          cfnContent.Resources.PermissionForEventsToInvokeLambda = {
+            Type: 'AWS::Lambda::Permission',
+            Properties: {
+              FunctionName: {
+                Ref: 'LambdaFunction',
+              },
+              Action: 'lambda:InvokeFunction',
+              Principal: 'events.amazonaws.com',
+              SourceArn: { 'Fn::GetAtt': ['CloudWatchEvent', 'Arn'] },
+            },
+          };
+          // append the outputs section of cloudwatchRULE
+          cfnContent.Outputs.CloudWatchEventRule = {
+            Value: {
+              Ref: 'CloudWatchEvent',
+            },
+          };
+          fs.writeFileSync(cfnFilePath, JSON.stringify(cfnContent, null, 4));
+        }
+      } catch (e) {
+        context.print.error(e.message);
+      }
+    }
+  } else {
+    if (await context.amplify.confirmPrompt.run('Do you want to update or remove the function schedule?', false)) {
+      const cfnContent = context.amplify.readJsonFile(cfnFilePath);
+      const scheduleEventOperationQuestion = {
+        type: 'list',
+        name: 'ScheduleEventOperation',
+        message: 'Select from the following options (Use arrow keys)',
+        choices: ['Update the schedule', 'Remove the schedule'],
+      };
+
+      const scheduleEventOperationAnswer = await inquirer.prompt([scheduleEventOperationQuestion]);
+
+      switch (scheduleEventOperationAnswer.ScheduleEventOperation) {
+        case 'Update the schedule': {
+          // add service walkthrough to get the cron expression
+          let cloudWatchRule = await cronServiceWalkthrough(context);
+          scheduleParams.cloudwatchEnabled = 'true';
+          scheduleParams.cloudwatchRule = cloudWatchRule;
+          fs.writeFileSync(cfnFilePath, JSON.stringify(cfnContent, null, 4));
+          break;
+        }
+        case 'Remove the schedule': {
+          scheduleParams.cloudwatchEnabled = 'false';
+          scheduleParams.cloudwatchRule = 'NONE';
+          delete cfnContent.Resources.CloudWatchEvent;
+          delete cfnContent.Resources.PermissionForEventsToInvokeLambda;
+          delete cfnContent.Outputs.CloudWatchEventRule;
+
+          fs.writeFileSync(cfnFilePath, JSON.stringify(cfnContent, null, 4));
+          break;
+        }
+        default:
+          console.log(`${scheduleEventOperationAnswer.scheduleEventOperation} not supported`);
+      }
+    }
+  }
+  return scheduleParams;
+}
+
+async function cronServiceWalkthrough(context: any) {
```

Comment:
I would break this function up into several other helper functions. Once you get into the switch statements it can be hard to follow which flow you're in

Reply:
fixed